Imports - Baseline XGBoost Model

In [1]:
# Core Libraries
import numpy as np
import joblib

# Model
from xgboost import XGBRegressor

# Evaluation Metrics
from sklearn.metrics import mean_absolute_error, mean_squared_error, r2_score

# Load Preprocessed Data

In [2]:
X_train = joblib.load(r"D:\Python\SmartSignalAI\data\X_train.pkl")
y_train = joblib.load(r"D:\Python\SmartSignalAI\data\y_train.pkl")

X_val   = joblib.load(r"D:\Python\SmartSignalAI\data\X_val.pkl")
y_val   = joblib.load(r"D:\Python\SmartSignalAI\data\y_val.pkl")

X_test  = joblib.load(r"D:\Python\SmartSignalAI\data\X_test.pkl")
y_test  = joblib.load(r"D:\Python\SmartSignalAI\data\y_test.pkl")

print("Data loaded successfully!")

Data loaded successfully!


# Train Basic XGBoost Model

In [3]:
xgb_model = XGBRegressor(
    n_estimators=200,
    learning_rate=0.1,
    max_depth=6,
    random_state=42,
    n_jobs=-1
)

xgb_model.fit(X_train, y_train)

print("XGBoost model training completed.")

XGBoost model training completed.


# Validation Evaluation

In [4]:
y_val_pred = xgb_model.predict(X_val)

mae_val  = mean_absolute_error(y_val, y_val_pred)
rmse_val = np.sqrt(mean_squared_error(y_val, y_val_pred))
r2_val   = r2_score(y_val, y_val_pred)

print("XGBoost - Validation Metrics")
print("MAE :", round(mae_val, 3))
print("RMSE:", round(rmse_val, 3))
print("R²  :", round(r2_val, 3))

XGBoost - Validation Metrics
MAE : 5.702
RMSE: 7.733
R²  : 0.499


# Test Evaluation

In [5]:
y_test_pred = xgb_model.predict(X_test)

mae_test  = mean_absolute_error(y_test, y_test_pred)
rmse_test = np.sqrt(mean_squared_error(y_test, y_test_pred))
r2_test   = r2_score(y_test, y_test_pred)

print("XGBoost - Test Metrics")
print("MAE :", round(mae_test, 3))
print("RMSE:", round(rmse_test, 3))
print("R²  :", round(r2_test, 3))

XGBoost - Test Metrics
MAE : 5.81
RMSE: 7.907
R²  : 0.507


Random Forest (Tuned) Test R² = 0.702
XGBoost Test R² = 0.507

# Save Model

In [ ]:
joblib.dump(
    xgb_model,
    r"D:\Python\SmartSignalAI\data\xgb_model.pkl"
)

print("XGBoost model saved successfully!")